provider/ultradns_test.go

```diff
@@ -0,0 +1,658 @@
+			Type: "A",
+		}
+
+		rrsets, _ = provider.client.RRSets.Select(rrsetKey)
+		assert.Equal(t, rrsets[0].RData, []string{"1.1.2.2", "192.168.0.24", "1.2.3.4"})
+
+		changes = &plan.Changes{}
+		changes.UpdateNew = []*endpoint.Endpoint{{DNSName: "kubernetes-ultradns-provider-test.com", Targets: endpoint.Targets{"1.1.2.2"}, RecordType: "A", RecordTTL: 100}}
+
+		err = provider.ApplyChanges(context.Background(), changes)
+		if err != nil {
+			t.Errorf("should not fail, %s", err)
+		}
+
+		rrsetKey = udnssdk.RRSetKey{
+			Zone: "kubernetes-ultradns-provider-test.com.",
+			Name: "kubernetes-ultradns-provider-test.com.",
+			Type: "A",
+		}
+
+		rrsets, _ = provider.client.RRSets.Select(rrsetKey)
+		assert.Equal(t, rrsets[0].RData, []string{"1.1.2.2"})
+
+		changes = &plan.Changes{}
+		changes.Delete = []*endpoint.Endpoint{{DNSName: "kubernetes-ultradns-provider-test.com", Targets: endpoint.Targets{"1.1.2.2", "192.168.0.24"}, RecordType: "A"}}
+
+		err = provider.ApplyChanges(context.Background(), changes)
+		if err != nil {
+			t.Errorf("should not fail, %s", err)
+		}
+
+		resp, _ := provider.client.Do("GET", "https://api.ultradns.com/zones/kubernetes-ultradns-provider-test.com./rrsets/A/kubernetes-ultradns-provider-test.com.", nil, udnssdk.RRSetListDTO{})
+		assert.Equal(t, resp.Status, "404 Not Found")
+
+	}
+}
+
+// Test case to check sbpool creation
+func TestUltraDNSProvider_newSBPoolObjectCreation(t *testing.T) {
+	mocked := mockUltraDNSRecord{nil}
+	mockedDomain := mockUltraDNSZone{nil}
+
+	provider := &UltraDNSProvider{
+		client: udnssdk.Client{
+			RRSets: &mocked,
+			Zone:   &mockedDomain,
+		},
+	}
+	sbpoolRDataList := []udnssdk.SBRDataInfo{}
+	changes := &plan.Changes{}
+	changes.UpdateNew = []*endpoint.Endpoint{{DNSName: "kubernetes-ultradns-provider-test.com.", Targets: endpoint.Targets{"1.1.2.2", "192.168.0.24"}, RecordType: "A", RecordTTL: 100}}
+	changesList := &UltraDNSChanges{
+		Action: "UPDATE",
+		ResourceRecordSetUltraDNS: udnssdk.RRSet{
+			RRType:    "A",
+			OwnerName: "kubernetes-ultradns-provider-test.com.",
+			RData:     []string{"1.1.2.2", "192.168.0.24"},
+			TTL:       100,
+		},
+	}
+
+	for _, _ = range changesList.ResourceRecordSetUltraDNS.RData {
+
+		rrdataInfo := udnssdk.SBRDataInfo{
+			RunProbes: true,
+			Priority:  1,
+			State:     "NORMAL",
+			Threshold: 1,
+			Weight:    nil,
+		}
+		sbpoolRDataList = append(sbpoolRDataList, rrdataInfo)
+	}
+	sbPoolObject := udnssdk.SBPoolProfile{
+		Context:     udnssdk.SBPoolSchema,
+		Order:       "ROUND_ROBIN",
+		Description: "kubernetes-ultradns-provider-test.com.",
+		MaxActive:   2,
+		MaxServed:   2,
+		RDataInfo:   sbpoolRDataList,
+		RunProbes:   true,
+		ActOnProbes: true,
+	}
+
+	actualSBPoolObject, _ := provider.newSBPoolObjectCreation(context.Background(), changesList)
+	assert.Equal(t, sbPoolObject, actualSBPoolObject)
+
+}
+
+//Testcase to check fail scenario for multiple AAAA targets
+func TestUltraDNSProvider_MultipleTargetAAAA(t *testing.T) {
```

Comment:
Remove this, as we will support RD pool for AAAA records.

Reply:
Done Appropriate addition of testcase